### PR TITLE
Transform to and from 'did:key' style keys in oob messages 

### PR DESCRIPTION
**Pull request by nrempel**
This pull request converts to and from `did:key` style keys to conform with https://github.com/hyperledger/aries-rfcs/tree/master/features/0360-use-did-key

### CODE REVIEW

File: aries_cloudagent/wallet/util.py [hunk 1]
```diff
@@ -57,3 +59,19 @@ def b58_to_bytes(val: str) -> bytes:
 def bytes_to_b58(val: bytes) -> str:
     """Convert a byte string to base 58."""
     return base58.b58encode(val).decode("ascii")
+
+
+def naked_to_did_key(key: str) -> str:
```
Comment (nrempel) [inline aries_cloudagent/wallet/util.py:64]: FYI @TelegramSam

File: aries_cloudagent/protocols/out_of_band/v1_0/manager.py [hunk 2]
```diff
@@ -142,8 +143,17 @@ async def create_invitation(
...
                 _type="did-communication",
-                recipient_keys=connection_invitation.recipient_keys,
-                routing_keys=connection_invitation.routing_keys,
+                recipient_keys=[
+                    naked_to_did_key(key)
+                    for key in connection_invitation.recipient_keys
+                ]
+                if connection_invitation.recipient_keys
+                else [],
+                routing_keys=[
```
Comment (sklump) [inline aries_cloudagent/protocols/out_of_band/v1_0/manager.py:152]: similarly,
```
routing_keys=naked_to_did_key(key) for key in connection_invitation.routing_keys or []]
```

File: aries_cloudagent/protocols/out_of_band/v1_0/manager.py [hunk 3]
```diff
@@ -224,6 +236,18 @@ async def receive_invitation(
                     "request block must be empty for invitation message type."
                 )
 
+            # Transform back to 'naked' verkey
+            service.recipient_keys = (
+                [did_key_to_naked(key) for key in service.recipient_keys]
```
Comment (sklump) [inline aries_cloudagent/protocols/out_of_band/v1_0/manager.py:241]:
```
[did_key_to_naked(key) for key in service.recipient_keys or []]
```

File: aries_cloudagent/protocols/out_of_band/v1_0/manager.py [hunk 4]
```diff
@@ -224,6 +236,18 @@ async def receive_invitation(
...
+                if service.recipient_keys
+                else []
+            )
+            service.routing_keys = (
+                [did_key_to_naked(key) for key in service.routing_keys]
```
Comment (sklump) [inline aries_cloudagent/protocols/out_of_band/v1_0/manager.py:246]:
```
[did_key_to_naked(key) for key in service.routing_keys or []]
```

**Comment by nrempel** [comment]
Thanks @sklump, cleaned up those issues

**Comment by sklump** [comment]
Note that there's samey code in attachment decorator: https://github.com/hyperledger/aries-cloudagent-python/blob/240366c9f1dd16c389dd55daec9a8ae4a49305d2/aries_cloudagent/messaging/decorators/attach_decorator.py#L209

I will wait for this PR to merge and then I think these conversions, plus something like `qual_key()` deserve a module somewhere. I'll take care of it once this one is merged.

**Pull request merged**